docs/learn/voting-dapp.mdx

```diff
@@ -575,33 +575,82 @@ if (!candidates) return;
 
 ```jsx
 async function vote() {
-  if (!walletClient || !address || number === "") {
-    console.error("Missing required data for voting");
-    return;
-  }
-  if (
-    isNaN(Number(number)) ||
-    Number(number) < 0 ||
-    Number(number) >= candidates.length
-  ) {
-    console.error("Invalid candidate index");
-    return;
-  }
-  if (!votingStatus) {
-    console.error("Voting is not currently active");
-    return;
-  }
   try {
-    console.log("Attempting to vote for candidate index:", number);
-    writeContract({
+    // Basic validation checks
+    if (!walletClient) {
+      console.error("Wallet client is not initialized");
+      throw new Error("Please connect your wallet");
+    }
+
+    if (!address) {
+      console.error("No wallet address found");
+      throw new Error("Please connect your wallet to vote");
+    }
+
+    if (number === "") {
+      console.error("No candidate number provided");
+      throw new Error("Please enter a candidate number");
+    }
+
+    const candidateNumber = Number(number);
+
+    // Enhanced number validation
+    if (isNaN(candidateNumber)) {
```

Comment:
There's a lot of lines of code for checking the `number`/`candidateNumber` variable

Can this be condensed to the following?

```js
if (parseInt(number) === NaN || parseInt(number) < 0 || parseInt(number) >= candidates.length) {
   console.error("Invalid candidate index");
   return;
}

```